src/lib.rs

```diff
@@ -197,8 +197,11 @@ pub enum ClientError {
     #[error("url parse error: {0}")]
     ParseUrl(#[from] url::ParseError),
```

Comment:
Would you like to wrap `ParseJson` and `ParseUrl` into a wrapper type named `ParseError` as you mentioned in #137 ?

Reply:
I honestly don't think it's necessary anymore. It will just make pattern matching more complicated